src/functionalTest/java/uk/gov/hmcts/reform/wacaseeventhandler/controllers/WarningEventHandlerControllerTest.java

```diff
@@ -0,0 +1,502 @@
+
+        String timeExtensionWarning = "[{\"warningCode\":\"Code101\","
+            + "\"warningText\":\"Warning Text 101\"}]";
+        WarningValues timeExtensionWarningValues = new WarningValues(timeExtensionWarning);
+
+        String followUpOverdueWarning = "[{\"warningCode\":\"Code102\","
+            + "\"warningText\":\"Warning Text 102\"}]";
+        WarningValues overdueWarningValues = new WarningValues(followUpOverdueWarning);
+
+        // check for warnings flag on both the tasks
+        assertTaskHasMultipleWarnings(caseIdForTask1, task1Id, timeExtensionWarningValues);
+        assertTaskHasMultipleWarnings(caseIdForTask1, task2Id, overdueWarningValues);
+
+        // task3Id should not contain any warnings because the category is not correlated
+        assertTaskWithoutWarnings(caseIdForTask1, task3Id, false);
+
+        // tear down all tasks
+        tearDownMultipleTasks(Arrays.asList(task1Id, task2Id, task3Id), "completed");
+    }
+
+    /**
+     * Scenario: 1 event on 2 different caseIds creates 2 different warnings each for all task types.
+     */
+    @Test
+    public void given_multiple_caseIDs_when_actions_is_warn_then_mark_all_tasks_with_warnings() {
+        //caseId1 with category Case progression
+        String caseId1 = UUID.randomUUID().toString();
+        String taskIdDmnColumn = "attendCma";
+        final String caseId1Task1Id = createTaskWithId(
+            caseId1,
+            "listCma",
+            "", "cmaListed", false,
+            taskIdDmnColumn
+        );
+
+        //caseId1 with category Case progression
+        String taskId2DmnColumn = "reviewRespondentResponse";
+        String caseId2 = UUID.randomUUID().toString();
+        final String caseId2Task1Id = createTaskWithId(caseId2, "uploadHomeOfficeAppealResponse",
+                                                       "", "respondentReview",
+                                                       false, taskId2DmnColumn);
+        // Then cancel all tasks on both caseIDs
+        sendMessage(caseId1, "_DUMMY_makeAnApplication",
+                    "", "", false);
+        waitSeconds(5);
+        sendMessage(caseId2, "_DUMMY_makeAnApplication",
+                    "", "", false);
+        waitSeconds(5);
+
+        String warningsAsJson = "[{\"warningCode\":\"TA01\","
+            + "\"warningText\":\"There is an application task which "
+            + "might impact other active tasks\"},"
+            + "{\"warningCode\":\"TA02\","
+            + "\"warningText\":\"There is another task on this case that needs your attention\"}"
+            + "]";
+        // check for warnings flag on both the tasks
+        assertTaskHasMultipleWarnings(caseId1, caseId1Task1Id, new WarningValues(warningsAsJson));
+        assertTaskHasMultipleWarnings(caseId2, caseId2Task1Id, new WarningValues(warningsAsJson));
+
+        // tear down all tasks
+        tearDownMultipleTasks(Arrays.asList(caseId1Task1Id, caseId2Task1Id), "completed");
+    }
+
```

Comment:
What about when the scenario to show where the warnings have different, they are appended?